moveit_experimental/moveit_jog_arm/src/jog_arm.cpp

```diff
@@ -238,18 +260,106 @@ bool JogInterfaceBase::readParameters(ros::NodeHandle& n)
   return true;
 }
 
+void JogArm::run()
+{
+  ros::Rate main_rate(1. / ros_parameters_.publish_period);
+  trajectory_msgs::JointTrajectory outgoing_command;
+
+  // Wait for low pass filters to stabilize
+  ROS_INFO_STREAM_NAMED(LOGNAME, "Waiting for low-pass filters to stabilize.");
+  ros::Duration(10 * ros_parameters_.publish_period).sleep();
+
+  while (ros::ok() && !shared_variables_.stop_requested)
+  {
+    // Check for stale cmds
+    {
+      const std::lock_guard<std::mutex> lock(latest_state_mutex_);
+      shared_variables_.command_is_stale =
+          ((ros::Time::now() - latest_command_stamp_) >= ros::Duration(ros_parameters_.incoming_command_timeout));
+    }
+
+    // Read the latest outgoing command out of the queue
+    popLatest(*outgoing_command_queue_, outgoing_command);
+
+    // Publish the most recent trajectory, unless the jogging calculation thread tells not to
+    if (shared_variables_.ok_to_publish)
+    {
+      // Put the outgoing msg in the right format
+      // (trajectory_msgs/JointTrajectory or std_msgs/Float64MultiArray).
+      if (ros_parameters_.command_out_type == "trajectory_msgs/JointTrajectory")
+      {
+        outgoing_command.header.stamp = ros::Time::now();
+        outgoing_cmd_pub_.publish(outgoing_command);
+      }
+      else if (ros_parameters_.command_out_type == "std_msgs/Float64MultiArray")
+      {
+        std_msgs::Float64MultiArray joints;
+        if (ros_parameters_.publish_joint_positions)
+          joints.data = outgoing_command.points[0].positions;
+        else if (ros_parameters_.publish_joint_velocities)
+          joints.data = outgoing_command.points[0].velocities;
+        outgoing_cmd_pub_.publish(joints);
+      }
+    }
+    else if (shared_variables_.command_is_stale)
+    {
+      ROS_WARN_STREAM_THROTTLE_NAMED(10, LOGNAME, "Stale command. "
+                                                  "Try a larger 'incoming_command_timeout' parameter?");
+    }
+    else
+    {
+      ROS_DEBUG_STREAM_THROTTLE_NAMED(10, LOGNAME, "All-zero command. Doing nothing.");
+    }
+
+    main_rate.sleep();
+  }
+}
+
+void JogArm::start()
```

Comment:
This function name could easily be confused with `run()`. I think you should change the name of one or the other. I think this makes sense:

run() --> processCommands()

Reply:
`run` is a common name for the main for a thread so I figured it was a good name.  `start`/`stop` are the public methods.

Reply:
I didn't notice `run` is private. OK, that makes sense.